torch/_inductor/ir.py

```diff
@@ -5101,10 +5101,19 @@ def codegen(self, wrapper):
         raw_args = [
             self.get_kwargs_value(k) for k in self.ordered_kwargs_for_cpp_kernel
         ]
+
+        # NOTE: raw_args doesn't include autotuned args.
+        # But, kernel.constexprs includes indices of autotuned args.
+        # So, let's recalculate constexpr indices wrt to raw_args.
+        constexpr_indices = []
+        for idx, kwarg in enumerate(self.ordered_kwargs_for_cpp_kernel):
+            if kernel.arg_names.index(kwarg) in kernel.constexprs:
```

Comment:
Why would `idx` not be equal to `kernel.arg_names.index(kwarg)`?

Reply:
That's because `self.ordered_kwargs_for_cpp_kernel` excludes autotuned params.

```
@triton.jit
def mykernel(
  param0,  param1,  param2,
  param3: tl.constexpr,   # autotuned
  param4,                 # non-constexpr
):
```

then ```self.ordered_kwargs_for_cpp_kernel: [param0, param1, param2, param4]```

Reply:
it gets populated here
https://github.com/pytorch/pytorch/blob/9dc4bd7466176ff667990c61681d651f042539ad/torch/_inductor/ir.py#L5147-L5150